### ISSUE TITLE
hyper-parameters to reproduce protonet results on resnet-12?

### ISSUE THREAD

**Issue by Tsingularity** [Sha-Lab/FEAT#43]
Hi thanks for the great work!

I went through the github issue and found someone asked the same question. However, seems like at different times, your response is also slightly different, sometimes lr=1e-3 and sometimes be 1e-4. So could you finalize the answer for the setting of init lr, schedule, gamma, max epoch and temperature? thanks

i have tried some combination of these hyper-parameters with the pre-trained model you provided. But for 5way 5shot i cannot reach 80. Also when i look at the curve of training acc/error and validation acc, i found they actually change very little through the whole training process and the best validation epoch is basically the first 5 epochs. Just wondering is this behaviour normal or similar to your end?

Thanks in advance!

**Comment by Han-Jia** [comment]
Please use the following command to get the results. In my experiments, loss value will be small at first with the help of the pre-trained weights, but the model with the best meta-val accuracy will be after 50 epochs.

python train_fsl.py  --max_epoch 200 --model_class FEAT  --backbone_class Res12 --dataset MiniImageNet --way 5 --eval_way 5 --shot 5 --eval_shot 5 --query 15 --eval_query 15 --balance 0.1 --temperature 64 --temperature2 32 --lr 0.0002 --lr_mul 10 --lr_scheduler step --step_size 40 --gamma 0.5 --gpu 0 --init_weights ./saves/initialization/miniimagenet/Res12-pre.pth --eval_interval 1 --use_euclidean

**Comment by Tsingularity** [comment]
thanks! let me take a try lol

**Comment by JimZAI** [comment]
> thanks! let me take a try lol



> Hi thanks for the great work!
> 
> I went through the github issue and found someone asked the same question. However, seems like at different times, your response is also slightly different, sometimes lr=1e-3 and sometimes be 1e-4. So could you finalize the answer for the setting of init lr, schedule, gamma, max epoch and temperature? thanks
> 
> i have tried some combination of these hyper-parameters with the pre-trained model you provided. But for 5way 5shot i cannot reach 80. Also when i look at the curve of training acc/error and validation acc, i found they actually change very little through the whole training process and the best validation epoch is basically the first 5 epochs. Just wondering is this behaviour normal or similar to your end?
> 
> Thanks in advance!

May I ask if the experimental phenomenon you mentioned above is related to ProtoNet method? I also I found this problem when i reproduced the ProtoNet. Or are you just talking about FEAT?

**Comment by Tsingularity** [comment]
> > thanks! let me take a try lol
> 
> > Hi thanks for the great work!
> > I went through the github issue and found someone asked the same question. However, seems like at different times, your response is also slightly different, sometimes lr=1e-3 and sometimes be 1e-4. So could you finalize the answer for the setting of init lr, schedule, gamma, max epoch and temperature? thanks
> > i have tried some combination of these hyper-parameters with the pre-trained model you provided. But for 5way 5shot i cannot reach 80. Also when i look at the curve of training acc/error and validation acc, i found they actually change very little through the whole training process and the best validation epoch is basically the first 5 epochs. Just wondering is this behaviour normal or similar to your end?
> > Thanks in advance!
> 
> May I ask if the experimental phenomenon you mentioned above is related to ProtoNet method? I also I found this problem when i reproduced the ProtoNet. Or are you just talking about FEAT?

I was talking about using their pre-trained weights to train protonet. but i also found similar phenomenon when i train the FEAT.

**Comment by JimZAI** [comment]
Thank you for your response!

Actually, my real concern is Issue #45. Please excuse my carelessness!



At 2020-11-17 04:59:28, "Luming Tang" <notifications@github.com> wrote:

thanks! let me take a try lol

Hi thanks for the great work!
I went through the github issue and found someone asked the same question. However, seems like at different times, your response is also slightly different, sometimes lr=1e-3 and sometimes be 1e-4. So could you finalize the answer for the setting of init lr, schedule, gamma, max epoch and temperature? thanks
i have tried some combination of these hyper-parameters with the pre-trained model you provided. But for 5way 5shot i cannot reach 80. Also when i look at the curve of training acc/error and validation acc, i found they actually change very little through the whole training process and the best validation epoch is basically the first 5 epochs. Just wondering is this behaviour normal or similar to your end?
Thanks in advance!

May I ask if the experimental phenomenon you mentioned above is related to ProtoNet method? I also I found this problem when i reproduced the ProtoNet. Or are you just talking about FEAT?

I was talking about using their pre-trained weights to train protonet. but i also found similar phenomenon when i train the FEAT.
